### PR TITLE
Add new ExternalShareActions API

### PR DESCRIPTION
**Pull request by skjnldsv**
See example: 

https://github.com/nextcloud/files_downloadlimit/pull/5
https://github.com/nextcloud/files_downloadlimit/blob/32ae1dbf2df12a262f849af07ef4c4ea8f7c3cfd/src/main.js
https://github.com/nextcloud/files_downloadlimit/blob/32ae1dbf2df12a262f849af07ef4c4ea8f7c3cfd/src/models/DownloadLimitAction.js

This API is a more global way of injecting entries into the link share menu.
It can be any type currently supported by the vue components.
In opposite of the old (and now legacy deprecated) API that only allowed links (for the social app).

### CODE REVIEW

**Review by julien-nc — COMMENTED** [review]
Looks good.
Too bad the `files_downloadlimit` example does not work (yet :grin:).
I'll make some real life tests by registering another component soon.

File: apps/files_sharing/src/services/ExternalShareActions.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+			|| typeof action.data !== 'function' // () => {disabled: true}
+			|| !Array.isArray(action.shareType) // [OC.Share.SHARE_TYPE_LINK, ...]
+			|| typeof action.handlers !== 'object' // {click: () => {}, ...}
+			|| !Object.values(action.handlers).every(handler => typeof handler === 'function')) {
+			console.error('Invalid action provided', action)
+			return false
+		}
+
+		// Check duplicates
+		const hasDuplicate = this._state.actions.findIndex(check => check.id === action.id) > -1
```
Comment (Pytal) [inline apps/files_sharing/src/services/ExternalShareActions.js:70]:
```suggestion
		const hasDuplicate = this._state.actions.some(check => check.id === action.id)
```
Might be more straightforward :)

**Comment by skjnldsv** [comment]
> Too bad the `files_downloadlimit` example does not work (yet ).

It does. Please check the dedicated pull request :)

File: apps/files_sharing/src/services/ExternalShareActions.js (hunk 1 above)
Comment (skjnldsv) [reply apps/files_sharing/src/services/ExternalShareActions.js:70]: Aaaah! I always forget about the new `some`! Awesome!

**Review by blizzz — APPROVED** [review]

**Review by Pytal — APPROVED** [review]
🚀🚀

**Comment by skjnldsv** [comment]
Rebased with latest `@nextcloud/vue`

**Pull request merged**